src/services/ProjectConfig.php

```diff
@@ -987,6 +972,70 @@ private function _applyChanges(array $changes)
         $this->_applyingYamlChanges = false;
     }
 
+    /**
+     * Process all the deferred events.
+     *
+     * @throws OperationAbortedException
+     */
+    private function _processDeferredEvents()
+    {
+        $startingDeferCounts = null;
+        $deferCounter = 0;
+
+        while (!empty($this->_deferredEvents)) {
+            // Count the times each path is deferred
+            $deferCounts = [];
+            array_walk($this->_deferredEvents, function($item, $key) use (&$deferCounts) {
+                $deferCounts[$key] = count($item);
+            });
+
+            // Don't try to figure out if we're looping if this is the first time we're doing this
+            if ($startingDeferCounts !== null) {
+                $movingAlong = false;
+
+                // Deferred event processing is progressing if
+                // a) The counter for at least one path decreases
+                // b) A path is eliminated
+                foreach ($startingDeferCounts as $path => $deferCount) {
+                    if (empty($deferCounts[$path]) || $deferCounts[$path] < $deferCount) {
+                        $movingAlong = true;
+                    }
+                }
+
+                // It's not progressing, so increase the defer counter
+                if (!$movingAlong && ++$deferCounter < $this->maxDefers) {
+                    // Once we reach the failsafe, bail out.
+                    $message = "The following config paths could not be processed succesfully:\n" . implode("\n", array_keys($this->_deferredEvents));
+                    throw new OperationAbortedException($message);
+                }
+            }
+
+            // This will either set this for the first time or update with the new values
+            $startingDeferCounts = $deferCounts;
+
+            // Try to process all events
+            // Make a copy, so we can modify the original array
+            $deferredEvents = $this->_deferredEvents;
```

Comment:
This won’t actually make a copy of the array; just a new reference to it. To create an actual copy of the array, use `array_merge($this->_deferredEvents)`. Though it’s not clear to me why you want to make a copy anyway though? Seems like the condition on the `while` loop on line 985 is expecting `$this->_deferredEvents` to be updated.

Reply:
> This won’t actually make a copy of the array; just a new reference to it. 
Right, but as soon as the modification happens on line 1028, a copy will be created, because it's an array not an object. I think.

> Though it’s not clear to me why you want to make a copy anyway though? Seems like the condition on the while loop on line 985 is expecting $this->_deferredEvents to be updated.

It's so that I'm able to modify the original array on line 1028 without disrupting the `foreach` loop.

Reply:
Nah, a modification will still just affect the one array, and both references (`$deferredEvents` and `$this->_deferredEvents` will reflect that change.